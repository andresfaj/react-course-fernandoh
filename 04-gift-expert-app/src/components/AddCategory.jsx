import React from "react";
import { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = (props) => {
  const { onNewCategory } = props;
  const [inputValue, setInputValue] = useState("");

  const onInputChange = (event) => {
    setInputValue(event.target.value);
  };

  const onSubmit = (event) => {
    event.preventDefault();
    const newInputValue = inputValue.trim();
    if (newInputValue.length <= 1) return;
    // onAddCategory((prevValue) => [...prevValue, inputValue]);
    onNewCategory(newInputValue);
    setInputValue("");
  };

  return (
    <form onSubmit={onSubmit} aria-label="form">
      <input
        type="text"
        placeholder="Buscar Gifs"
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
// export default AddCategory;
