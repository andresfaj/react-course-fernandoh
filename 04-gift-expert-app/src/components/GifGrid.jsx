import PropTypes from "prop-types";
import { GiftItem } from "./GiftItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs({ category });
  return (
    <>
      <h3>{category}</h3>
      {isLoading && <h2>Cargando...</h2>}
      <div className="card-grid">
        {images.map(({ id, title, url }) => (
          <GiftItem key={id} title={title} url={url} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};

// export default GifGrid;
