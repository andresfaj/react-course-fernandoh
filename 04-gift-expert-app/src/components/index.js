export * from "./AddCategory";
export * from "./GifGrid";
export * from "./GiftItem";
