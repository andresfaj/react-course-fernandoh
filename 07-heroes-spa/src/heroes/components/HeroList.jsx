import { getHeroesByPublisher } from "../helpers/getHeroesByPublisher";
import { HeroCard } from "./HeroCard";

export const HeroList = (props) => {
  const { publisher } = props;

  const heroes = getHeroesByPublisher(publisher);

  return (
    <div className="row rows-cols-1 row-cols-md-3 g-3">
      {heroes.map((heroe) => (
        <HeroCard key={heroe.id} {...heroe} />
      ))}
    </div>
  );
};
