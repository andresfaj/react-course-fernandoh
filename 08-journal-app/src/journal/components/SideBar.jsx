import { TurnedInNot } from "@mui/icons-material";
import {
  Box,
  Divider,
  Drawer,
  Grid,
  List,
  ListItem,
  ListItemButton,
  ListItemIcon,
  ListItemText,
  Toolbar,
  Typography,
} from "@mui/material";

export const SideBar = ({ drawerWidth }) => {
  return (
    <Box
      component="nav"
      sx={{ width: { sm: drawerWidth }, flexShrink: { sm: 0 } }}
    >
      <Drawer
        variant="permanent" //temporary
        open
        sx={{
          display: { xs: "block" },
          "& .MuiDrawer-paper": { boxSizing: "border-box", width: drawerWidth },
        }}
      >
        <Toolbar>
          <Typography variant="h6" noWrap component="div">
            Andres Achury
          </Typography>
        </Toolbar>
        <Divider />

        <List>
          {["Enero", "Febrero", "Marzo", "Abril"].map((month) => {
            return (
              <ListItem key={month} disablePadding>
                <ListItemButton>
                  <ListItemIcon>
                    <TurnedInNot />
                  </ListItemIcon>
                  <Grid container>
                    <ListItemText primary={month} />
                    <ListItemText
                      secondary={"Deserunt occaecat culpa Lorem ex."}
                    />
                  </Grid>
                </ListItemButton>
              </ListItem>
            );
          })}
        </List>
      </Drawer>
    </Box>
  );
};
