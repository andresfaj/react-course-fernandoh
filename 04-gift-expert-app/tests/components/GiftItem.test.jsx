import { render, screen } from "@testing-library/react";
import { GiftItem } from "../../src/components/GiftItem";

describe("Test on GiftItem Component", () => {
  const title = "Test item";
  const url = "https://photos.com/";
  test("Must match with the snapshot", () => {
    const { container } = render(<GiftItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("Must show the image with url and Alt", () => {
    render(<GiftItem title={title} url={url} />);
    // screen.debug();
    // expect(screen.getByRole("img").src).toBe(url);
    // expect(screen.getByRole("img").alt).toBe(title);
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test("Must show the title", () => {
    render(<GiftItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
