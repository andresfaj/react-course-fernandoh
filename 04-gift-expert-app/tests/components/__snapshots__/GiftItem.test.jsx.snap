// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on GiftItem Component Must match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Test item"
      src="https://photos.com/"
    />
    <p>
      Test item
    </p>
  </div>
</div>
`;
